### ISSUE TITLE
Unable to run rake assets:precompile without an available database when Hyrax v2.1.0.beta1 gem installed

### ISSUE THREAD

**Issue by blackrat** [samvera/hyrax#2785]
### Descriptive summary

V2.1.0.beta1 doesn't allow rake tasks to be run without a database connection.

Initialisation now contains a block for `config.after_initialize` that calls `Hyrax.persist_registered_roles!` which in turn does a `Sipity::Role.find_or_create_by!` and hence requires database access at initialisation time.

### Rationale

In a Dockerized environment, you, for example, want to run:

`rake assets:precompile RAILS_ENV=production`

outside the container to generate production assets

### Expected behavior

Assets are precompiled before the container starts up.

### Actual behavior

`rake aborted!
PG::ConnectionBad: could not translate host name "database_container_name" to address: nodename nor servname provided, or not known`

### Steps to reproduce the behavior

You don't need to have Docker to reproduce the behaviour, just a database adapter and a non-accessible database detailed in your `database.yml` file.
Create a database.yml file which contains:
```
production:
  adapter: postgresql
  host: database_container_name
  port: 3232
  username: wibble
  password: wobble
```
The full details don't really matter as long as it's a database adapter that doesn't create a local database when run (like sqlite3).

Once this is in place, run:

`rake assets:precompile RAILS_ENV=production`

**Comment by no-reply** [comment]
`RAILS_ENV=production SECRET_KEY_BASE='fakesecret1234' DB_ADAPTER=nulldb DATABASE_URL='postgresql://fake' bundle exec rake assets:precompile` is a work-around

**Comment by no-reply** [comment]
the specific issue mentioned might be fixed by `d02ac4082cfcb3980d55f5ae45b74e4226cfd9f3`

**Comment by no-reply** [comment]
after testing, yes i think this issue as such was fixed some time ago.

i'm pretty sure the remaining issues are generic rails issues, where a config and/or `DB_ADAPTER=nulldb` need to be present to do `Rails.application.initialize!` which runs when precompiling assets.
